### PR TITLE
feat: warehouse explorer 

### PR DESCRIPTION
**Pull request by jordienr**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

---
to try it:
- go to Logs & Analytics
- click on the source dropdown and pick warehouse
- (if you don't have a collection create one in the sidebar)
- edit the query or use the Template dropdown to populate the editor with a query for one of your collections
- Run the query and you should see the events
- optional: If you don't have any events you can go to the collection in the sidebar and copypaste the curl from the dialog to send some events

feat: Allows users (with warehouse flag active) to query and explore their collections from the logs explorer.
ui: Updated sidebar titles to make it warehouse friendly.
chore: Updated api path on warehouse queries.
fixes: fixed access token form and updated to new input components.

### CODE REVIEW

File: apps/studio/components/interfaces/DataWarehouse/CreateWarehouseAccessToken.tsx [hunk 1]
```diff
@@ -61,27 +55,32 @@ const CreateWarehouseAccessToken = ({ onSubmit, loading, open, setOpen }: Create
         visible={open}
         alignFooter="right"
         loading={loading}
+        onConfirm={() => {
+          form.handleSubmit((data) => {
+            onSubmit(data)
+          })()
```
Comment (joshenlim) [inline apps/studio/components/interfaces/DataWarehouse/CreateWarehouseAccessToken.tsx:61]: is the `()` needed here?

File: apps/studio/components/interfaces/Settings/Logs/LogsQueryPanel.tsx [hunk 2]
```diff
@@ -14,28 +7,39 @@ import {
...
-  IconChevronDown,
   IconClipboard,
   IconExternalLink,
   IconX,
   Popover,
   SidePanel,
   Tabs,
 } from 'ui'
+
+import Link from 'next/link'
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Logs/LogsQueryPanel.tsx:19]: nit: just fyi for consistency we're following a writing style where we separate our import statements as such:

```
// Example:
<external imports>
import Link from 'next/link'

<internal imports>
import DatePickers from './Logs.DatePickers'
```

File: apps/studio/pages/project/[ref]/logs/explorer/index.tsx [hunk 3]
```diff
@@ -208,20 +306,44 @@ export const LogsExplorerPage: NextPageWithLayout = () => {
...
           <ShimmerLine active={isLoading} />
-          <CodeEditor
-            id={editorId}
-            language="pgsql"
-            defaultValue={editorValue}
-            onInputChange={(v) => setEditorValue(v || '')}
-            onInputRun={handleRun}
-          />
+          {sourceType === 'warehouse' ? (
+            <CodeEditor
```
Comment (joshenlim) [inline apps/studio/pages/project/[ref]/logs/explorer/index.tsx:331]: Command Enter doesn't run the code for warehouse code editor

Comment (joshenlim) [reply apps/studio/pages/project/[ref]/logs/explorer/index.tsx:331]: Not sure if BE or FE issue but when I do try to run the query by clicking the Run button, it's saying that i don't have a collection of that name, although I do
image

File: apps/studio/components/interfaces/Settings/Logs/LogsQueryPanel.tsx [hunk 4]
```diff
@@ -44,86 +48,171 @@ export interface LogsQueryPanelProps {
...
-                  .sort((a, b) => a.label!.localeCompare(b.label!))
-                  .map((template) => (
+            {dataSource === 'warehouse' && (
+              <DropdownMenu>
+                <DropdownMenuTrigger asChild>
+                  <Button type="default" iconRight={<ChevronDown />}>
+                    Templates
+                  </Button>
+                </DropdownMenuTrigger>
+                <DropdownMenuContent>
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Logs/LogsQueryPanel.tsx:137]: We should probably add a scroll area for this one when theres > n collections (can refer to ProjectDropdown)
image

Comment (joshenlim) [reply apps/studio/components/interfaces/Settings/Logs/LogsQueryPanel.tsx:137]: also lets do align="start", consistent with the logs as data source

File: apps/studio/components/interfaces/Settings/Logs/Warehouse.utils.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  query: string
+  name: string
+  description: string
+}
+
+export function createWarehouseQueryTemplates(
+  collections: { name: string }[]
+): WarehouseQueryTemplate[] {
+  return collections.map((collection) => ({
+    query:
```
Comment (joshenlim) [inline apps/studio/components/interfaces/Settings/Logs/Warehouse.utils.ts:11]: reference to https://github.com/supabase/supabase/pull/27611/files#r1666784496

not sure if BE or FE issue but this query doesnt seem to be working: "Cant find source {source name}"

File: apps/studio/next-env.d.ts [hunk 6]
```diff
@@ -1,6 +1,5 @@
 /// <reference types="next" />
 /// <reference types="next/image-types/global" />
-/// <reference types="next/navigation-types/compat/navigation" />
```
Comment (joshenlim) [inline apps/studio/next-env.d.ts:3]: just to double check this is intentional?

File: apps/studio/pages/project/[ref]/logs/explorer/index.tsx [hunk 7]
```diff
@@ -130,9 +179,49 @@ export const LogsExplorerPage: NextPageWithLayout = () => {
...
+
     if (value && typeof value === 'string') {
       setEditorValue(value)
     }
+
+    if (sourceType === 'warehouse') {
+      const whQuery = warehouseEditorValue
+
+      if (!warehouseCollections?.length) {
+        toast.error("You don't have any collections in your warehouse yet.")
```
Comment (joshenlim) [inline apps/studio/pages/project/[ref]/logs/explorer/index.tsx:191]: lets avoid contractions for consistency
```suggestion
        toast.error("You do not have any collections in your warehouse yet.")
```

File: apps/studio/pages/project/[ref]/logs/explorer/index.tsx [hunk 8]
```diff
@@ -130,9 +179,49 @@ export const LogsExplorerPage: NextPageWithLayout = () => {
...
+        return
+      }
+
+      // Check that the user is not trying to query logs tables and warehouse collections at the same time
+      const logsSources = Object.values(LOGS_TABLES)
+      const logsSourceExists = logsSources.find((source) => whQuery.includes(source))
+
+      if (logsSourceExists) {
+        toast.error(
+          'Cannot query logs tables from a warehouse query. Please remove the logs table from the query.'
```
Comment (joshenlim) [inline apps/studio/pages/project/[ref]/logs/explorer/index.tsx:212]: Does this imply that the warehouse name cannot match the name of any of the logs sources?
if yes then perhaps we should add some client side validation when creating collections

**Review by joshenlim — APPROVED** [review]
Left some comments but approving either way to prevent the long running PRs - and also since this feature is not public yet so no issue

File: apps/studio/pages/project/[ref]/logs/explorer/index.tsx (hunk 8 above)
Comment (jordienr) [reply apps/studio/pages/project/[ref]/logs/explorer/index.tsx:212]: Actually you could, I was thinking about what would happen if users tried to join or query logs tables and wh collections, but wh collections can have the same name so this check is not the best way to do it. I'll remove it for now and see if I come up with something better, this was just for UX.

**Pull request merged**